spec.md

```diff
@@ -1663,6 +1716,509 @@ message GetCapacityResponse {
 
 If the plugin is unable to complete the GetCapacity call successfully, it MUST return a non-ok gRPC code in the gRPC status.
 
+#### `CreateVolumeGroup`
+
+**ALPHA FEATURE**
+
+This RPC will be called by the CO to create a new volume group on behalf of a user. This operation MUST be idempotent. If a volume group corresponding to the specified volume group name already exists, is compatible with the specified parameters in the CreateVolumeGroupRequest, the Plugin MUST reply 0 OK with the corresponding CreateVolumeGroupResponse. CSI Plugins MAY create the following types of volume groups:
+
+Create a new empty volume group.
+After the empty group is created, create a new volume, specifying the group name in the volume.
+
+At restore time, create a single volume from individual snapshot and then join an existing group.
+Create an empty group.
+Create a volume from snapshot, specifying the group name in the volume.
+
+Future goals:
+Create a new volume group from a source group snapshot.
+Create a new volume group from a source group.
+Create a new volume group and add a list of existing volumes to the group.
+
+Non goal:
+Create a new group and at the same time create a list of new volumes in the group.
+
+```protobuf
+message CreateVolumeGroupRequest {
+  option (alpha_message) = true;
+
+  // suggested name for volume group (required for idempotency)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // params passed from VolumeGroupClass
+  // This field is OPTIONAL.
+  map<string, string> parameters = 2;
+
+  // Secrets required by plugin to complete volume group creation
+  // request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 3 [(csi_secret) = true];
+}
+
+message CreateVolumeGroupResponse {
+  option (alpha_message) = true;
+
+  // Contains all attributes of the newly created volume group.
+  // This field is REQUIRED.
+  VolumeGroup volume_group = 1;
+}
+
+message VolumeGroup {
+  option (alpha_message) = true;
+
+  // The identifier for this volume group, generated by the plugin.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
```

Comment:
I would suggest adding the Name field assigned the VolumeGroup as part of the VolumeGroup response, especially since the Name field in CreateVolumeGroupRequest is only "suggested", so it's impossible to know definitively what the Name turned out to be. However I compared this with the Volume response and it doesn't return the Name either so you are consistent.